src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/calculator/PriceCalculator.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.pecs.jpc.calculator
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.Location
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.LocationRepository
+import uk.gov.justice.digital.hmpps.pecs.jpc.pricing.Price
+import uk.gov.justice.digital.hmpps.pecs.jpc.pricing.PriceRepository
+import uk.gov.justice.digital.hmpps.pecs.jpc.pricing.Supplier
+import uk.gov.justice.digital.hmpps.pecs.jpc.reporting.Journey
+import uk.gov.justice.digital.hmpps.pecs.jpc.reporting.MoveFilterer
+import uk.gov.justice.digital.hmpps.pecs.jpc.reporting.MovePersonJourneysEvents
+
+@Component
+class PriceCalculatorFactory(
+        @Autowired val locationRepository: LocationRepository,
+        @Autowired val priceRepository: PriceRepository) {
+
+    fun calculator(moves: List<MovePersonJourneysEvents>): PriceCalculator {
+        val nomisAgencyId2Location = locationRepository.findAll().associateBy(Location::nomisAgencyId)
+        val (sercoPrices, geoPrices) = priceRepository.findAll().partition { it.supplier == Supplier.SERCO }
```

Comment:
Not sure.  Perhaps load the prices in the `fun standardPrices(supplier: Supplier): List<MovePrice>` when needed. Dunno.

Reply:
I did previously do that but I felt it was unclear when things were initialised / loaded in.